src/components/common/Button.tsx

```diff
@@ -141,20 +134,18 @@ const Button = ({
   fullWidth,
   disabled,
   loading,
-  href,
```

Comment:
If you destructure props already before typescript knows what it's type is, it is unsure whether that is even a valid prop for this component. So before you use any of those 'dynamic' props that aren't always present, you need to do a check like `"href" in props` or `typeof props.href !=="undefined"` (or in this case I named the rest properties `buttonAttributes` instead of `props` since that's a bit more declarative.